From 9afa120576c69268e5eee6336a55447b2b428588 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Thu, 22 Jan 2015 17:29:18 +0800
Subject: [PATCH 2/4] MA-6133 Add missed av stop event response

Added the av stop event response on the idle
state.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 btif/src/btif_av.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/btif/src/btif_av.c b/btif/src/btif_av.c
index 5715748..a477072 100644
--- a/btif/src/btif_av.c
+++ b/btif/src/btif_av.c
@@ -343,6 +343,20 @@ static BOOLEAN btif_av_state_idle_handler(btif_sm_event_t event, void *p_data)
             btif_rc_handler(event, (tBTA_AV*)p_data);
             break;
 
+        case BTIF_AV_STOP_STREAM_REQ_EVT:
+            if (btif_av_cb.peer_sep == AVDT_TSEP_SNK)
+            {
+              /* immediately flush any pending tx frames while suspend is pending */
+              btif_a2dp_set_tx_flush(TRUE);
+            }
+            if (btif_av_cb.peer_sep == AVDT_TSEP_SRC)
+            {
+                btif_a2dp_set_rx_flush(TRUE);
+            }
+
+            btif_a2dp_on_stopped(NULL);
+            break;
+
         case BTA_AV_RC_CLOSE_EVT:
             if (tle_av_open_on_rc.in_use) {
                 BTIF_TRACE_DEBUG("BTA_AV_RC_CLOSE_EVT: Stopping AV timer.");
-- 
1.8.0

